opentelemetry-config/src/providers.rs

```diff
@@ -0,0 +1,227 @@
+//! OpenTelemetry SDKs using declarative YAML configurations.
+
+mod meter_provider;
+
+use std::collections::HashMap;
+
+use opentelemetry::KeyValue;
+use opentelemetry_sdk::{
+    logs::SdkLoggerProvider, metrics::SdkMeterProvider, trace::SdkTracerProvider, Resource,
+};
+
+use crate::{
+    model::Telemetry, providers::meter_provider::MeterProvider, ConfigurationProviderRegistry,
+    ProviderError,
+};
+
+/// Holds the configured telemetry providers
+pub struct TelemetryProviders {
+    meter_provider: Option<SdkMeterProvider>,
+    logs_provider: Option<SdkLoggerProvider>,
+    traces_provider: Option<SdkTracerProvider>,
+}
+
+impl TelemetryProviders {
+    pub fn new(
+        meter_provider: Option<SdkMeterProvider>,
+        logs_provider: Option<SdkLoggerProvider>,
+        traces_provider: Option<SdkTracerProvider>,
+    ) -> Self {
+        TelemetryProviders {
+            meter_provider,
+            logs_provider,
+            traces_provider,
+        }
+    }
+
+    /// Returns a reference to the configured MeterProvider, if any
+    pub fn meter_provider(&self) -> Option<&SdkMeterProvider> {
+        self.meter_provider.as_ref()
+    }
+
+    /// Returns a reference to the configured LoggerProvider, if any
+    pub fn logs_provider(&self) -> Option<&SdkLoggerProvider> {
+        self.logs_provider.as_ref()
+    }
+
+    /// Returns a reference to the configured TracerProvider, if any
+    pub fn traces_provider(&self) -> Option<&SdkTracerProvider> {
+        self.traces_provider.as_ref()
+    }
+
+    /// Configures the Telemetry providers based on the provided configuration
+    fn configure(
+        configuration_registry: &ConfigurationProviderRegistry,
+        config: &Telemetry,
+    ) -> Result<TelemetryProviders, ProviderError> {
+        let resource: Resource = Self::as_resource(&config.resource);
+
+        let meter_provider_option: Option<SdkMeterProvider>;
+        if let Some(metrics_config) = &config.metrics {
+            let mut meter_provider_builder =
+                SdkMeterProvider::builder().with_resource(resource.clone());
+            meter_provider_builder = MeterProvider::configure(
+                &configuration_registry.metrics,
+                meter_provider_builder,
+                metrics_config,
+            )?;
+            let meter_provider = meter_provider_builder.build();
+            meter_provider_option = Some(meter_provider);
+        } else {
+            meter_provider_option = None;
+        }
+
+        // TODO: Add logs configuration
+        let logs_provider_option = None;
+
+        // TODO: Add traces configuration
+        let traces_provider_option = None;
+
+        let providers = TelemetryProviders::new(
+            meter_provider_option,
+            logs_provider_option,
+            traces_provider_option,
+        );
+
+        Ok(providers)
+    }
+
+    /// Configures the Telemetry providers from a YAML string
+    pub fn configure_from_yaml(
```

Comment:
Let's add `_str` as a suffix to make the method name more descriptive.

```suggestion
    pub fn configure_from_yaml_str(
```